zds/member/views.py

```diff
@@ -1087,15 +1088,23 @@ def modify_karma(request):
     note = KarmaNote()
     note.user = profile.user
     note.staff = request.user
-    note.comment = request.POST["warning"]
+    note.comment = request.POST.get("warning", "")
+
     try:
         note.value = int(request.POST["points"])
     except (KeyError, ValueError):
         note.value = 0
 
-    note.save()
-
-    profile.karma += note.value
-    profile.save()
+    try:
+        if note.comment == "":
+            raise ValueError("note.comment must not be empty")
+        elif note.value > 100 or note.value < -100:
+            raise ValueError("note.value must be between -100 and 100 {} given".format(note.value))
+        else:
+            note.save()
+            profile.karma += note.value
+            profile.save()
+    except ValueError as e:
+        logging.getLogger("zds.member").warn("A bad form was sent for karma {}".format(e))
```

Comment:
D'accord. La réaction de @gustavi confirme mon impression que c'est pas explicite, et celle d' @artragis que c'est un message destiné à un être humain et donc censé être compréhensible. :)

Je propose ceci :

```diff
-        logging.getLogger("zds.member").warn("A bad form was sent for karma {}".format(e))
+        logging.getLogger("zds.member").warn("ValueError: modifying karma failed because {}".format(e))
```